Libraries/Text/RCTShadowText.m

```diff
@@ -119,7 +119,7 @@ - (NSTextStorage *)buildTextStorageForWidth:(CGFloat)width
 
   NSTextContainer *textContainer = [NSTextContainer new];
   textContainer.lineFragmentPadding = 0.0;
-  textContainer.lineBreakMode = _numberOfLines > 0 ? NSLineBreakByTruncatingTail : NSLineBreakByClipping;
```

Comment:
Here's what I added:

```js
if (Platform.OS === 'ios' && newProps.lineBreakMode === undefined) {
  // Prevent `this.props` from being mutated.
  if (newProps === this.props) {
    newProps = { ...this.props };
  }
  // If 'newProps.numberOfLines' is undefined, 'clipping' is used.
  newProps.lineBreakMode = newProps.numberOfLines > 0 ? 'truncating-tail' : 'clipping';
}
```

[Link to mentioned code](https://github.com/aleclarson/react-native/blob/lineBreakMode-PR/Libraries/Text/Text.js#L222-L227).